### PR TITLE
Add Civic

### PR DESCRIPTION
**Pull request by onursumer**
Related to genome-nexus/genome-nexus#301

Collapsed (default) view:

clinical_implications_collapsed

Expanded view:

clinical_implications_expanded

### CODE REVIEW

File: src/component/variantPage/TherapeuticImplication.tsx [hunk 1]
```diff
@@ -1,220 +1,43 @@
...
+import OncoKb from './therapeuticImplication/OncoKb';
+import Civic from './therapeuticImplication/Civic';
 
 interface ITherapeuticImplicationProps {
     oncokb: IndicatorQueryResp | undefined;
+    civic?: ICivicVariantIndex;
     isCanonicalTranscriptSelected: boolean;
 }
 
-export const SENSITIVE_LEVELS = [
```
Comment (onursumer) [inline src/component/variantPage/TherapeuticImplication.tsx:19]: added a new component under `src/component/variantPage/therapeuticImplication/OncoKb.tsx` and moved OncoKB related code there.

File: src/component/variantPage/therapeuticImplication/Civic.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            overlay={
+                <span>
+                    <a
+                        href={props.url}
+                        target="_blank"
+                        rel="noopener noreferrer"
+                    >
+                        CIViC
+                    </a>{' '}
+                    is TODO
```
Comment (onursumer) [inline src/component/variantPage/therapeuticImplication/Civic.tsx:26]: We need a brief description of CIViC here

File: src/component/variantPage/therapeuticImplication/OncoKb.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    'LEVEL_3',
+    'LEVEL_3A',
+    'LEVEL_3B',
+    'LEVEL_4',
+];
+
+export const RESISTANT_LEVELS = ['LEVEL_R1', 'LEVEL_R2'];
+
+@observer
+class OncoKb extends React.Component<IOncoKbProps> {
```
Comment (onursumer) [inline src/component/variantPage/therapeuticImplication/OncoKb.tsx:32]: The code here is not new. Moved from `src/component/variantPage/TherapeuticImplication.tsx`.

**Comment by leexgh** [comment]
Not sure if there is a good way to show, sometimes `Disease` is empty but `Drugs` `Significance` and `Level` are the same:

image

**Comment by onursumer** [comment]
@leexgh Thanks for testing this!

An empty disease column might actually be a bug because in the OncoKB website all the rows have the cancer type information: https://www.oncokb.org/gene/BRCA1/E23Vfs*17. Looks like OncoKB combines those levels & drugs in a different / smarter way.

In case of no evidence, I think it is better to hide the table and show N/A or something like that.

**Comment by leexgh** [comment]
@onursumer This is awesome! Thanks!

A few comments and questions:
1. Search box can't search by level
https://deploy-preview-86--genome-nexus-frontend.netlify.app/variant/7:g.140453136A%3ET
image

2. Should we also do "Collapsed" vs "Expanded" design for exon and functional prediction(those have a table in tooltip)?

3. For example this variant `17:g.7578534C>G`, on cbioportal it has civic in annotation column, but on genome nexus it is N/A:
https://www.cbioportal.org/results/mutations?tab_index=tab_visualize&Action=Submit&session_id=611ac7c9e4b015b63e9fa769&plots_horz_selection=%7B%7D&plots_vert_selection=%7B%7D&plots_coloring_selection=%7B%7D
https://deploy-preview-86--genome-nexus-frontend.netlify.app/variant/17:g.7578534C%3EG 
image

**Comment by inodb** [comment]
@onursumer 

Really nice work and quick turnaround!

- [ ] The collapse button seems slightly misaligned with the evidence text. Is there some way to center the fonts?

    Screen Shot 2021-08-16 at 6 21 44 PM

Other notes:

- I'm actually slightly confused about this BRCA1 frameshift: https://deploy-preview-86--genome-nexus-frontend.netlify.app/variant/17:g.41276045_41276046del. Is it correct that there is no civic result? Seems like it's the same in cBioPortal. It does show an evidence somehow for `R1443*` there. Ok to ignore for now but just wanted to write it down

To add on to one of @leexgh's  discussion items

> 2. Should we also do "Collapsed" vs "Expanded" design for exon and functional prediction(those have a table in tooltip)?

Good point. I think for "functional prediction" the use case is slightly different in that it's more of a legend rather than a data table. Would be good to redesign tho but prolly ok to do in separate PR. The other one that would be good to have collapse/expand is gnomad, since it seems very similar to the use case in oncockb/civic. @leexgh which one did you mean by exon?

**Comment by leexgh** [comment]
@inodb oops I mean gnomad

**Comment by onursumer** [comment]
@leexgh Thanks for reviewing! See my inline comments below.

> @onursumer This is awesome! Thanks!
> 
> A few comments and questions:
> 
> 1. Search box can't search by level
>    https://deploy-preview-86--genome-nexus-frontend.netlify.app/variant/7:g.140453136A%3ET
>    image

I haven't enabled search by level, since it is a single character. It might be still useful for OncoKB because the levels are numerical, but it wouldn't really work for Civic because the levels are letters.
> 2. Should we also do "Collapsed" vs "Expanded" design for gnomad and functional prediction(those have a table in tooltip)?

Yes, we should. But probably in a follow up PR. We need some refactoring for that. We directly use some components from cbioportal frontend packages and they are designed as table cells with a default tooltip.
> 3. For example this variant `17:g.7578534C>G`, on cbioportal it has civic in annotation column, but on genome nexus it is N/A:
>    https://www.cbioportal.org/results/mutations?tab_index=tab_visualize&Action=Submit&session_id=611ac7c9e4b015b63e9fa769&plots_horz_selection=%7B%7D&plots_vert_selection=%7B%7D&plots_coloring_selection=%7B%7D
>    https://deploy-preview-86--genome-nexus-frontend.netlify.app/variant/17:g.7578534C%3EG
>    image

I think this is expected. If you look at the tooltip in cbioportal you will see there are no `K132N Entries` listed. This is consistent with https://civicdb.org/events/genes/45/summary/variants/1113/summary#variant. There are no evidences listed for this variant. Do you think `N/A` is confusing? Should we display `No supporting evidences` or something like that?

**Comment by onursumer** [comment]
@inodb Thanks for reviewing! 

> @onursumer
> 
> Really nice work and quick turnaround!
> 
> * [ ]  The collapse button seems slightly misaligned with the evidence text. Is there some way to vertically align the fonts?
>     Screen Shot 2021-08-16 at 6 21 44 PM

Right, this is because we use a button instead of an actual link, so the styling is a little off. Our linter is not happy with an `<a>` element without an `href`, but I guess I can find a way to fix this.

> 
> Other notes:
> 
> * I'm actually slightly confused about this BRCA1 frameshift: https://deploy-preview-86--genome-nexus-frontend.netlify.app/variant/17:g.41276045_41276046del. Is it correct that there is no civic result? Seems like it's the same in cBioPortal. It does show an evidence somehow for `R1443*` there. Ok to ignore for now but just wanted to write it down

Good point. We are reusing the logic implemented in the cbioportal frontend. There might be something wrong with the way we query civic for variants. We can further investigate and improve the existing logic if necessary.
> 
> To add on to one of @leexgh's discussion items
> 
> > 1. Should we also do "Collapsed" vs "Expanded" design for exon and functional prediction(those have a table in tooltip)?
> 
> Good point. I think for "functional prediction" the use case is slightly different in that it's more of a legend rather than a data table. Would be good to redesign tho but prolly ok to do in separate PR. The other one that would be good to have collapse/expand is gnomad, since it seems very similar to the use case in oncockb/civic. @leexgh which one did you mean by exon?

Right, we should probably deal with other resources in a follow up PR.

**Comment by jjgao** [comment]
>> 3. For example this variant `17:g.7578534C>G`, on cbioportal it has civic in annotation column, but on genome nexus it is N/A:
>>    https://www.cbioportal.org/results/mutations?tab_index=tab_visualize&Action=Submit&session_id=611ac7c9e4b015b63e9fa769&plots_horz_selection=%7B%7D&plots_vert_selection=%7B%7D&plots_coloring_selection=%7B%7D
>>    https://deploy-preview-86--genome-nexus-frontend.netlify.app/variant/17:g.7578534C%3EG
>>    image

> I think this is expected. If you look at the tooltip in cbioportal you will see there are no `K132N Entries` listed. This is consistent with https://civicdb.org/events/genes/45/summary/variants/1113/summary#variant. There are no evidences listed for this variant. Do you think `N/A` is confusing? Should we display `No supporting evidences` or something like that?

I think N/A is fine when there is no evidence. Is it a bug on the cBioPortal side to show the CIViC icon?

**Comment by onursumer** [comment]
> @onursumer looks really nice! All great points, @leexgh @inodb
> 
> A tiny comment: the expanding/collapsing icons are different from the ones in PTM track - should unify them?

Either way is fine to me. Not related, but I still think that track labels are too small in the Genome Nexus frontend.
> 
> > > 1. For example this variant `17:g.7578534C>G`, on cbioportal it has civic in annotation column, but on genome nexus it is N/A:
> > >    https://www.cbioportal.org/results/mutations?tab_index=tab_visualize&Action=Submit&session_id=611ac7c9e4b015b63e9fa769&plots_horz_selection=%7B%7D&plots_vert_selection=%7B%7D&plots_coloring_selection=%7B%7D
> > >    https://deploy-preview-86--genome-nexus-frontend.netlify.app/variant/17:g.7578534C%3EG
> > >    image
> 
> > I think this is expected. If you look at the tooltip in cbioportal you will see there are no `K132N Entries` listed. This is consistent with https://civicdb.org/events/genes/45/summary/variants/1113/summary#variant. There are no evidences listed for this variant. Do you think `N/A` is confusing? Should we display `No supporting evidences` or something like that?
> 
> I think N/A is fine when there is no evidence. Is it a bug on the cBioPortal side to show the CIViC icon?

Currently we check if the variant exits in Civic to decide whether to show the icon. We don't really check if there are any evidences. We still have a link out (https://civicdb.org/events/genes/45/summary/variants/1113/summary#variant) to the variant in the tooltip. So, maybe this is not a bug but a feature?

**Review by leexgh — APPROVED** [review]

**Pull request merged**